src/cookieConsent.js

```diff
@@ -0,0 +1,191 @@
+import React, { useEffect } from "react";
+import { Button, Box, Snackbar, Dialog, DialogTitle, DialogContent, DialogActions, makeStyles, Grid } from "@material-ui/core";
+import { DAY_IN_MS, MONTH_IN_MS } from "./utils/TimeUtils";
+import { clearAnalytics, initAnalytics } from "./utils/analytics";
+import PropTypes from "prop-types";
+import { Link } from "react-router-dom";
+import { RouterLink } from "./utils";
+import { useMobile } from "./utils/media-queries";
+import useToggle from "./hooks/useToggle";
+
+export const COOKIE_ACCEPT_LIFETIME_MONTHS = 6;
+export const COOKIE_REJECT_LIFETIME_DAYS = 5;
+
+const useStyles = (isMobile) => makeStyles((theme) => ({
+    dialog: {
+        padding: theme.spacing(2),
+    },
+    buttonsArea: {
+        display: "flex",
+        justifyContent: "flex-end",
+        padding: theme.spacing(2),
+    },
+    snackbar: isMobile ? ({
+        "& .MuiSnackbarContent-action": {
+            paddingLeft: 0,
+        },
+    }) : ({}),
+}));
+
+const CookiesUsageDialog = ({ open, handleAccept, handleReject, handleToggle }) => {
+    const classes = useStyles()();
+    const isMobile = useMobile();
+
+    return (
+        <Dialog
+            open={open}
+            onClose={handleToggle}
+            classes={{ paper: classes.dialog }}
+        >
+            <DialogTitle>
+                Why do we use cookies?
+            </DialogTitle>
+            <DialogContent style={{ textAlign: "justify" }}>
+                <p>
+                    {"NIJobs uses cookies to collect usage to improve our product."}
+                </p>
+                <p>
+                    {"Regarding privacy issues, we simply collect search parameters and other \
+                        useful metrics without linking them to your identity."}
+                </p>
+                <p>
+                    {"For more information, consult our "}
+                    <Link component={RouterLink} to="privacy">
+                        {"Privacy Policy"}
+                    </Link>
+                    {"!"}
+                </p>
+                <p>
+                    {"Thank you!"}
+                </p>
+            </DialogContent>
+            <DialogActions className={classes.buttonsArea}>
+                <Grid container spacing={2} justifyContent={!isMobile ? "flex-end" : "center"}>
+                    <Grid item xs={12} sm="auto" marginRight={2}>
+                        <Button onClick={handleReject}>Use only essential cookies</Button>
+                    </Grid>
+                    <Grid item xs={12} sm="auto">
+                        <Button color="primary" variant="contained" onClick={handleAccept}>Accept</Button>
+                    </Grid>
+                </Grid>
+            </DialogActions>
+        </Dialog>
+    );
+};
+
+export const CookieConsent = () => {
+    const [open, toggleOpen, setClosed] = useToggle(false);
+
+    const [showWhyModal, toggleWhyModal, setClosedWhyModal] = useToggle(false);
```

Comment:
This is more consistent with the `useToggle` behavior. If we had a default value of `true` it would do the opposite.


```suggestion
    const [showWhyModal, toggleWhyModal, resetWhyModalOpeningState] = useToggle(false);
```